### PR TITLE
Feature/pipeline

### PR DESCRIPTION
**Pull request by slawler**

### CODE REVIEW

**Comment by ar-siddiqui** [comment]
@nick-j-roberts can you summarize here what this PR doing. Something like this:

This PR: 

	- does this
	- and that
	- and that

**Comment by nick-j-roberts** [comment]
@ar-siddiqui Please see below
Contributions of @ericjasonking:
- Provide sample .env file
- Add HMS grid conversion script

My contributions:
- Update function for sending jobs to batch (send-job.py) to use JSON documents as input rather than hardcoding values in script
- Update meilisearch upload process to be contained in functions and have if __name__ == "__main__" entry point
- Refactor s3 fetch functions to use pagination
- Create meilisearch function to update single attribute of existing documents in index
- Updated meilisearch to accept JSON input used in batch job submission rather than hardcoding values in script
- Updated sys.argv command line argument parsing to argparse for maintaining expected input data types and ease of use
- Added some docstrings and usage notes
- Revised requirements.txt to more fully represent dependencies of docker image (tested in container)

File: extract_storms_v2.py [hunk 1]
```diff
@@ -1,60 +1,99 @@
 import boto3
```
Comment (ar-siddiqui) [inline extract_storms_v2.py]: A docstring for the file at the top is really helpful to understand what this file does.

File: batch/send-job.py [hunk 2]
```diff
@@ -1,47 +1,146 @@
...
+
+    load_dotenv(find_dotenv())
+
+    logging.basicConfig(
+        level=logging.INFO,
+        format='{"time":"%(asctime)s", "level": "%(levelname)s", "message":%(message)s}',
+        handlers=[logging.StreamHandler()],
+    )
+
+    send_inputs = load_inputs("records/duwamish.json")
```
Comment (ar-siddiqui) [inline batch/send-job.py]: Can we remove this hardcoded value and make it a CLI argument.

File: batch/send-job.py [hunk 3]
```diff
@@ -1,47 +1,146 @@
...
+    print(send_inputs)
```
Comment (ar-siddiqui) [inline batch/send-job.py]: Delete this line.

File: batch/send-job.py [hunk 4]
```diff
@@ -1,47 +1,146 @@
...
+        "-a",
+        job_input.atlas_14_uri,
+        "-sm",
+        str(job_input.scale),
+    ]
+    return cmd_list
+
+
+def send(job_input: JobInput) -> None:
+    # Create batch client
```
Comment (ar-siddiqui) [inline batch/send-job.py:126]: These kinds of comments are not super helpful as they just explain what you are doing in the code, not why you are doing it.

File: batch/batch-logs.py [hunk 5]
```diff
@@ -3,34 +3,48 @@
 import json
```
Comment (ar-siddiqui) [inline batch/batch-logs.py:3]: You could use vs code to sort your import to follow PEP-8 convention.

File: ms/create-hms-grid.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
+import os
```
Comment (ar-siddiqui) [inline ms/create-hms-grid.py:2]: Add a docstring explaining what file does.

File: ms/meilisearch-upload.py [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+    from dotenv import find_dotenv, load_dotenv
+
+    logging.basicConfig(
+        level=logging.INFO,
+        format='{"time":"%(asctime)s", "level": "%(levelname)s", "message":%(message)s}',
+        handlers=[logging.StreamHandler()],
+    )
+
+    load_dotenv(find_dotenv())
+    ms_inputs = load_inputs("records/duwamish.json")
```
Comment (ar-siddiqui) [inline ms/meilisearch-upload.py]: Can we not do without hardcoding this?

File: ms/meilisearch-upload.py [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+                raise ValueError(
+                    f"Document id parsed for update does not exist in the database. ID searched: {restructured['id']}"
+                )
+    ms_client.index(INDEX).update_documents(docs)
+
+
+def build_index():
+    # build meilisearch index
+    ms_client = Client(os.environ["REACT_APP_MEILI_HOST"], api_key=os.environ["REACT_APP_MEILI_MASTER_KEY"])
+    ms_client.index(INDEX).delete()
```
Comment (ar-siddiqui) [inline ms/meilisearch-upload.py]: Do not silently delete the index.

File: ms/meilisearch-upload.py [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+            if ms_client.index(INDEX).get_document(restructured["id"]):
+                docs.append({"id": restructured["id"], update_attribute: restructured[update_attribute]})
+            else:
+                raise ValueError(
+                    f"Document id parsed for update does not exist in the database. ID searched: {restructured['id']}"
+                )
+    ms_client.index(INDEX).update_documents(docs)
+
+
+def build_index():
```
Comment (ar-siddiqui) [inline ms/meilisearch-upload.py]: I think the code for index creation should go to a separate file.

File: logger.py [hunk 10]
```diff
@@ -9,7 +9,10 @@ def set_up_logger(filename: str = None):
     logging.getLogger("aiobotocore").setLevel(logging.WARNING)
 
     logger = logging.getLogger()
-    logger.removeHandler(logger.handlers[0])
+    try:
```
Comment (ar-siddiqui) [inline logger.py]: Can you use if here and not do try except.

**Review by ar-siddiqui — APPROVED** [review]
I am approving since no major issues but see if you can address review comments.
